src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -37,14 +46,53 @@ def get(self, request, organization):
                 "id", flat=True
             )
 
+        teams = request.GET.getlist("team", [])
+        team_filter_query = None
+        if teams:
+            unassigned = None
+            if "unassigned" in teams:
+                teams.remove("unassigned")
+                unassigned = Q(owner_id=None)
+
+            if "myteams" in teams:
+                teams.remove("myteams")
+                for t in Team.objects.filter(
+                    id__in=OrganizationMemberTeam.objects.filter(
+                        organizationmember__in=OrganizationMember.objects.filter(
+                            user=request.user, organization_id=organization.id
+                        ),
+                        is_active=True,
+                    ).values("team")
+                ):
+                    teams.append(t.id)
+
+            for team_id in teams:  # Verify each Team id is numeric
+                if type(team_id) is not int and not team_id.isdigit():
+                    return Response(
+                        f"Invalid Team ID: {team_id}", status=status.HTTP_400_BAD_REQUEST
+                    )
+
+            teams = Team.objects.filter(id__in=set(map(int, teams)))
+            for team in teams:  # Verify user has access to each team
+                if not team.has_access(request.user):
+                    return Response(
+                        f"Error: You do not have permission to access {team.name}",
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
```

Comment:
We can do `request.has_team_access` for each team. Saves us running an extra query. We should also probably only do this for passed team ids, rather than the ones we already validated from `myteams`

Reply:
Nice, cleaner code